editor/src/clj/editor/collection.clj

```diff
@@ -817,6 +819,33 @@
   (let [ext->embedded-component-resource-type (workspace/get-resource-type-map workspace)]
     (collection-string-data/string-encode-collection-desc ext->embedded-component-resource-type collection-desc)))
 
+(defn- add-dropped-resource
+  [selection collection transform-props resource]
+  (let [ext (str/lower-case (resource/ext resource))
+        base-name (resource/base-name resource)]
+    (case ext
+      "go"
+      (let [id (gen-instance-id collection base-name)
```

Comment:
Unfortunately, doing this here will result in duplicate ids when dragging multiple objects into the scene at once if their base name matches. For example, if I drag both `/inventory/controller.go` and `/menu/controller.go` into the scene at once, It results in two objects named `controller`, but I would expect `controller` and `controller1`.

Reply:
Nice catch! Introduced `outline/name-resource-pairs` to handle that.